src/Mutator/Util/Result/SingleMutationResult.php

```diff
@@ -0,0 +1,36 @@
+<?php
+/**
+ * Copyright © 2017-2018 Maks Rafalko
+ *
+ * License: https://opensource.org/licenses/BSD-3-Clause New BSD License
+ */
+
+declare(strict_types=1);
+
+namespace Infection\Mutator\Util\Result;
+
+use PhpParser\Node;
+
+/**
+ * @internal
+ */
+final class SingleMutationResult implements MutatorResult
+{
+    /**
+     * @var Node|Node[]
+     */
+    private $node;
+
+    /**
+     * @param Node|Node[] $node
+     */
+    public function __construct($node)
```

Comment:
This doesn't have a typehint, as a 'single' mutation can also return an array of nodes to replace a node.

I could also split it up in an `ArrayMutationResult` or something like that.